### PR TITLE
[Pick 2.0 var](planner) enbale read data opt on origin planner

### PR DESCRIPTION
**Pull request by airborne12**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by eldenmoon** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java [hunk 1]
```diff
@@ -1128,6 +1131,9 @@ public void setMaxJoinNumberOfReorder(int maxJoinNumberOfReorder) {
                     + " use a skiplist to optimize the intersection."})
     public int invertedIndexConjunctionOptThreshold = 1000;
 
+    @VariableMgr.VarAttr(name = ENABLE_INDEX_DATA_READ_OPTIMIZED_ON_ORIGINAL_PLANNER, needForward = true)
```
Comment (xiaokang) [inline fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java:1134]: needForward = false

File: fe/fe-core/src/main/java/org/apache/doris/planner/OriginalPlanner.java [hunk 2]
```diff
@@ -491,19 +491,69 @@ private void pushSortToOlapScan() {
...
     private void pushOutColumnUniqueIdsToOlapScan(PlanFragment rootFragment, Analyzer analyzer) {
         Set<Integer> outputColumnUniqueIds = new HashSet<>();
-        // add '-1' to avoid the optimization incorrect work with OriginalPlanner,
-        // because in the storage layer will skip this optimization if outputColumnUniqueIds contains '-1',
-        // to ensure the optimization only correct work with nereidsPlanner
-        outputColumnUniqueIds.add(-1);
+        if (ConnectContext.get().getSessionVariable().enableIndexDataReadOptOnOrigPlanner) {
+            ArrayList<Expr> outputExprs = rootFragment.getOutputExprs();
+            for (Expr expr : outputExprs) {
+                if (expr instanceof SlotRef) {
```
Comment (xiaokang) [inline fe/fe-core/src/main/java/org/apache/doris/planner/OriginalPlanner.java:497]: process expr recursively to add all SlotRef to outputColumnUniqueIds

**Review by xiaokang — APPROVED** [review]
LGTM

**Pull request merged**